src/discord-cluster-manager/cogs/leaderboard_cog.py

```diff
@@ -24,159 +25,49 @@
-        )
-    except discord.errors.NotFound as e:
-        print(f"Webhook not found: {e}")
-        await send_discord_message(interaction, "❌ The webhook was not found.")
-
-    message_contents = [msg.content async for msg in github_thread.history(limit=None)]
-
-    # Compute eval or submission score, call runner here.
-    # TODO: Make this more robust later
-    score = extract_score("".join(message_contents))
-
-    with bot.leaderboard_db as db:
-        db.create_submission(
-            {
-                "submission_name": script.filename,
-                "submission_time": datetime.now(),
-                "leaderboard_name": leaderboard_name,
-                "code": submission_content,
-                "user_id": interaction.user.id,
-                "submission_score": score,
-                "gpu_type": gpu,
-            }
-        )
-
-    user_id = (
-        interaction.user.global_name if interaction.user.nick is None else interaction.user.nick
-    )
-
-    await send_discord_message(
-        interaction,
-        f"Successfully ran on {gpu} using GitHub runners!\n"
-        + f"Leaderboard '{leaderboard_name}'.\n"
-        + f"Submission title: {script.filename}.\n"
-        + f"Submission user: {user_id}.\n"
-        + f"Runtime: {score:.9f} seconds.",
-        ephemeral=True,
-    )
-
-
 class LeaderboardSubmitCog(app_commands.Group):
     def __init__(self, bot):
         super().__init__(name="submit", description="Submit to leaderboard")
         self.bot = bot
 
-    async def interaction_check(self, interaction: discord.Interaction) -> bool:
-        if interaction.channel_id != self.bot.leaderboard_submissions_id:
-            await interaction.response.send_message(
-                f"Please use submission commands in <#{self.bot.leaderboard_submissions_id}>",
-                ephemeral=True,
-            )
-            return False
-        return True
-
-    # Parent command that defines global options
-    @app_commands.describe(
-        leaderboard_name="Name of the competition / kernel to optimize",
-        script="The Python / CUDA script file to run",
-    )
-    @app_commands.autocomplete(leaderboard_name=leaderboard_name_autocomplete)
-    # TODO: Modularize this so all the write functionality is in here. Haven't figured
-    # a good way to do this yet.
-    async def submit(
-        self,
-        interaction: discord.Interaction,
-        leaderboard_name: str,
-        script: discord.Attachment,
-    ):
-        pass
-
-    @app_commands.command(name="modal", description="Submit leaderboard data for modal")
-    @app_commands.autocomplete(leaderboard_name=leaderboard_name_autocomplete)
-    async def submit_modal(
+    async def async_submit_cog_job(
         self,
         interaction: discord.Interaction,
         leaderboard_name: str,
         script: discord.Attachment,
+        command,
+        reference_code,
+        submission_content,
+        cog: commands.Cog,
+        gpu: str,
```

Comment:
string typed inputs is not infromative, you can make this an enum as well

Reply:
good idea